Use to_edge_lower_and_transform for XNNPack

jackzhxng opened the pull request:
### Summary
Use `to_edge_transform_and_lower` in `export_llama` for XNNPack. As part of these changes, this also means that you cannot specify multiple backends in `export_llama` in the args, although I'm not sure if that is happening anywhere at the moment.

Closes https://github.com/pytorch/executorch/issues/8621

Performance regression benchmarking for xnnpack (on android) vs. past 3 days:
<img width="1427" alt="Screenshot 2025-02-24 at 11 39 52 AM" src="https://github.com/user-attachments/assets/1640cf2c-a579-491f-8940-7ccfbe464903" />

These benchmark numbers also normally fluctuate a bit across runs and these differences are within the usual fluctuation ranges.

### Test plan
See if CI passes

iseeyuan commented:
Have you tested the performance of a model (like llama 3B), before and after? Asking because export_llama is used by different users to prepare for the .pte files in cpu. Please make sure there's no perf regress.

tarun292 commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -674,47 +674,91 @@ def _validate_args(args):
+        # pyre-fixme[16]: Module `backends` has no attribute `qualcomm`.
+        canonicalize_program(builder.edge_manager.exported_program())
+
+    return builder.to_executorch(passes=additional_passes)
```
Hmm there's a return in the middle of the function?

tarun292 commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -674,47 +674,91 @@ def _validate_args(args):
+    )
+    if args.verbose:
+        print_delegation_info(builder.edge_manager.exported_program().graph_module)
+    if args.num_sharding > 0 and args.qnn:
```
This code shouldn't be here right?

jackzhxng replied:
Oh yeah technically should remove since qnn. Will remove after benchmarking finishes

jackzhxng commented:
@iseeyuan please see the performance benchmark graph I posted in the pr description

mergennachin commented:
Can you run internal CI tests before merging? Otherwise looks good.

jackzhxng committed:
Use to_edge_lower_and_transform for XNNPack (#8624)

Summary:
Use `to_edge_transform_and_lower` in `export_llama` for XNNPack. As part of these changes, this also means that you cannot specify multiple backends in `export_llama` in the args, although I'm not sure if that is happening anywhere at the moment.

Closes https://github.com/pytorch/executorch/issues/8621

Performance regression benchmarking for xnnpack (on android) vs. past 3 days:
<img width="1427" alt="Screenshot 2025-02-24 at 11 39 52 AM" src="https://github.com/user-attachments/assets/1640cf2c-a579-491f-8940-7ccfbe464903" />

These benchmark numbers also normally fluctuate a bit across runs and these differences are within the usual fluctuation ranges.


Test Plan: See if CI passes

Differential Revision: D70124742

Pulled By: jackzhxng
